pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -71,7 +71,9 @@ func (p *Probe) checkUpdated() error {
 		return err
 	}
 
-	if s.CDSUpdates > 0 && s.LDSUpdates > 0 {
+	CDSUpdated := s.CDSUpdatesSuccess > 0 || s.CDSUpdatesRejection > 0
+	LDSUpdated := s.LDSUpdatesSuccess > 0 || s.LDSUpdatesRejection > 0
```

Comment:
how will this work, if envoy rejects all  updates from the very beginning?

Reply:
@rshriram I suppose the probe will signal 'all good' while the envoy is actually non-functional. That is certainly not nice, but the probe was introduced during 1.1 development and it breaks custom gateway certs, as they are not mounted on sidecars but DestinationRules get applied mesh-wide.

This is meant as a short-term mitigation, there are longer-term solutions discussed in https://github.com/istio/istio/issues/8541

Reply:
Also, it is a ready probe not a health probe. If everything is rejected, it"s ready, right? Nothing will change anymore.